### PR TITLE
Add support for query parameters in RunProxy()

### PR DESCRIPTION
**Pull request by rcknight**
I've been using this library as a simple proxy using the `RunProxy` extension, and noticed it doesn't forward the query string from the original request on to the target. 

This is easy enough to work around in `BeforeSend` with something like:
```
BeforeSend = (context, request) =>
  {
    request.RequestUri = new Uri(request.RequestUri.ToString() + context.Request.QueryString);
    return Task.CompletedTask;
  },
```
But I thought it would be nice if this just worked out of the box! Is there a reason for not forwarding the query string that i'm missing? If not, this PR adds that functionality.

I've worked against your v4 branch, which I hope was the right thing to do - and had to tweak the test HttpBoomerang a little so that I can assert on the uri as well as content.

All the unit tests pass but I have a bunch of failing integration tests when running locally, i've got a feeling that the third party service you use for those is at fault rather than anything i've changed!

### CODE REVIEW

**Comment by twitchax** [comment]
Nice!

I had a note to fix this, so this is cool.  I will take a look soon.

File: src/Test/RunProxy/RunProxyIntegrationTests.cs [hunk 1]
```diff
@@ -70,7 +70,7 @@ public async Task CanDoWebSockets()
 
             // HTTP test.
             var response = await _httpClient.PostAsync("http://localhost:5003/at/some/path", new StringContent(send1));
-            Assert.Equal(expected1, await response.Content.ReadAsStringAsync());
+            Assert.EndsWith(expected1, await response.Content.ReadAsStringAsync());
```
Comment (twitchax) [inline src/Test/RunProxy/RunProxyIntegrationTests.cs]: Why the change to `EndsWith`?

Comment (twitchax) [reply src/Test/RunProxy/RunProxyIntegrationTests.cs]: Oh, my bad.  I see.

**Comment by rcknight** [comment]
Thanks for the prompt review! :)

**Comment by twitchax** [comment]
Yep!
